server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -471,7 +471,14 @@ public void executeQueryPhase(InternalScrollSearchRequest request,
                                   SearchShardTask task,
                                   ActionListener<ScrollQuerySearchResult> listener) {
         final LegacyReaderContext readerContext = (LegacyReaderContext) findReaderContext(request.contextId(), request);
-        final Releasable markAsUsed = readerContext.markAsUsed(getScrollKeepAlive(request.scroll()));
+        final Releasable markAsUsed;
+        try {
+            markAsUsed = readerContext.markAsUsed(getScrollKeepAlive(request.scroll()));
+        } catch (Exception e) {
+            // We need to release the reader context of the scroll when we hit any exception (here the keep_alive can be too large)
+            processFailure(readerContext, e);
```

Comment:
can you replace the call with `freeReaderContext(context.id())`?

Reply:
++. I pushed https://github.com/elastic/elasticsearch/pull/62179/commits/3517cb56f1130288b92b9587e65e267fa6873626